### ISSUE TITLE
CodeAnalysis: CA3075: error for correct case

### ISSUE THREAD

**Issue by alex1815** [dotnet/roslyn-analyzers#1880]
**Version Used**: 
Microsoft.CodeAnalysis.FxCopAnalyzer 2.6.1, StyleCop.Analyzer 1.0.2
**Steps to Reproduce**:
After adding analyzer to existing code I got error below:

![image](https://user-images.githubusercontent.com/11064818/47850585-d32da280-dde6-11e8-8f08-274664bb3635.png)

I used suggestion from [docs](https://docs.microsoft.com/en-ca/visualstudio/code-quality/ca3075-insecure-dtd-processing?view=vs-2015#solution-4), but it didn't help.

**Expected Behavior**:
No errors
**Actual Behavior**:
CA3075	XmlTextReader instance created with insecure default settings.

Could you please clarify this behavior or mark it as bug otherwise.

**Comment by mavasani** [comment]
This is dupe of https://github.com/dotnet/roslyn-analyzers/issues/1591, which is assigned to @dotpaul. Paul, are you planning to address this soon? There have been multiple reports now about the false positives in CA3075 due to lack of flow analysis.

**Comment by alex1815** [comment]
As it's dupe, I'll close the issue, but still waiting estimate from @dotpaul. Thanks

**Comment by dotpaul** [comment]
Hi @alex1815, does your code happen to be targeting .NET Framework v4.5.1 or below?

For flow analysis fixes, that will require dataflow analysis support which is will be available in the next major release of the analyzers after v2.6.  I unfortunately don't have an estimate yet when I'll be able to complete all of it, as there a lot of scenarios covered by the XML rules.  But it's on my list!

I'll take a look today and tomorrow to see if there's something that can be fixed without flow analysis.

**Comment by dotpaul** [comment]
@alex1815 try this:
```csharp
var textReader = new XmlTextReader(sourceDataFilePath)
{
    DtdProcessing = DtdProcessing.Prohibit,
    XmlResolver = null
};
```

**Comment by alex1815** [comment]
@dotpaul Thanks. I'll try in Monday

**Comment by MadnessSoftware** [comment]
Getting CA3075 XML Error from generated code. VS should be creating error free auto-generated code and doesn't conflict with the analyzer.

[App.Designer.zip](https://github.com/dotnet/roslyn-analyzers/files/5029473/App.Designer.zip)

Been 2 years since this issue was detected.
Please advise.

**Comment by sharwell** [comment]
Duplicate of #1591
